department-of-veterans-affairs/abd-vro
# MCP-2147 - Enforce required ClaimCondition fields to be NotNull

## What was the problem?


- Execute the automatedClaim endpoint without diagnosticCode parameter in the request body. You will see the response code: 200. 
- Note:  If the "disabilityActionType" and  "disabilityClassificationCode" is not in the request body also giving the response code: 200 instead of 400
  - UPDATE: Kalai says this is expected behavior, so  nothing will be done to those here
- Expected response code : 400 

Associated tickets or Slack threads:

- [MCP-2147](https://amida.atlassian.net/browse/MCP-2147)

## How does this fix it?[^1]

- We now enforce, for `MasAutomatedClaimPayload`'s child Object `ClaimCondition` that the following fields be enforced as REQUIRED via `@NotNull`
  - `diagnosticCode`
  - ~~`disabilityActionType`~~
  - ~~`disabilityClassificationCode`~~

## How to test this PR
- Step 1: Submit a POST request to `/v2/automatedClaim` with a payload that omits `claimDetail.conditions.diagnosticCode` ~~and/or `claimDetail.conditions.disabilityActionType` and/or `claimDetail.conditions.disabilityClassificationCode`in any combination~~
  - Verify you receive a `400 Bad Request` with a message indicating the field(s) cannot be null
- Step 2: Submit a valid POST request to `/v2/automatedClaim` with a valid payload
  - Verify you receive a `200 OK`


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
